pwndbg/commands/cymbol.py

```diff
@@ -0,0 +1,269 @@
+gnu gcc compiled toolchain for your target architecture.
+
+You are advised to configure the 'cymbol-editor' config parameter to the path of your
+favorite text editor. Otherwise cymbol exapnds $EDITOR and $VISUAL environement varialbes
+to find the path to the default text editor.
+"""
+
+import argparse
+import os
+import subprocess
+import sys
+import tempfile
+
+import gdb
+
+import pwndbg
+import pwndbg.commands
+import pwndbg.gdblib.arch
+import pwndbg.lib.gcc
+import pwndbg.lib.tempfile
+from pwndbg.color import message
+
+gcc_compiler_path = pwndbg.gdblib.config.add_param(
+    "gcc-compiler-path",
+    "",
+    "Path to your own gnu gcc/g++ toolchain for generating imported symbols.",
+)
+
+cymbol_editor = pwndbg.gdblib.config.add_param(
+    "cymbol-editor", "", "Path to your editor of your choice for editing custom structures."
+)
+
+
+def OnlyWhenStructureExists(func):
+    def wrapper(custom_structure_name):
+        pwndbg_cachedir = pwndbg.lib.tempfile.cachedir("custom-symbols")
+        pwndbg_custom_structure_path = os.path.join(pwndbg_cachedir, custom_structure_name) + ".c"
+
+        if not os.path.exists(pwndbg_custom_structure_path):
+            print(message.error("No custom structure was found with the given name!"))
+            return
+
+        return func(custom_structure_name, pwndbg_custom_structure_path)
+
+    return wrapper
+
+
+def PromptForOverwrite(func):
+    def wrapper(custom_structure_name):
+        pwndbg_cachedir = pwndbg.lib.tempfile.cachedir("custom-symbols")
+        pwndbg_custom_structure_path = os.path.join(pwndbg_cachedir, custom_structure_name) + ".c"
+
+        if not os.path.exists(pwndbg_custom_structure_path):
+            return func(custom_structure_name)
+
+        option = input(
+            message.notice(
+                "A custom structure was found with the given name, would you like to overwrite it? [y/n] "
+            )
+        )
+
+        if option != "y":
+            return
+
+        return func(custom_structure_name)
+
+    return wrapper
+
+
+def generate_debug_symbols(custom_structure_path, pwndbg_debug_symbols_output_file=None):
+    if not pwndbg_debug_symbols_output_file:
+        _, pwndbg_debug_symbols_output_file = tempfile.mkstemp(prefix="custom-", suffix=".dbg")
+
+    # -fno-eliminate-unused-debug-types is a handy gcc flag that lets us extract debug symbols from non-used defined structures.
+    gcc_extra_flags = [
+        custom_structure_path,
+        "-c",
+        "-g",
+        "-fno-eliminate-unused-debug-types",
+        "-o",
+        pwndbg_debug_symbols_output_file,
+    ]
+    gcc_flags = None
+
+    if gcc_compiler_path != "":
+        if pwndbg.gdblib.arch.ptrsize == 8:
+            gcc_flags = [gcc_compiler_path, "-m64"]
+        elif pwndbg.gdblib.arch.ptrsize == 4:
+            gcc_flags = [gcc_compiler_path, "-m32"]
+    else:
+        # Should we check for exceptions here?
+        gcc_flags = pwndbg.lib.gcc.which(pwndbg.gdblib.arch)
```

Comment:
Not an exception, but you should probably check if the first element of the list is `None` (we should really change this method to just return None if it doesn't find the compiler and not return a list).